src/v/pandaproxy/schema_registry/test/compatibility_avro.cc

```diff
@@ -534,16 +526,107 @@ const absl::flat_hash_set<incompatibility> backward_expected{
    "expected: someEnum1 (alias resolution is not yet fully supported)"},
 };
 
-const auto compat_data = std::to_array<compat_test_data<incompatibility>>({
+struct compat_test_case {
+    std::string reader;
+    std::string writer;
+    absl::flat_hash_set<incompatibility> expected;
+};
+
+const auto compat_data = std::to_array<compat_test_case>({
```

Comment:
nit:
I find the color test cases a bit verbose. To help with readability, i would suggest:
- Parameterize  the reader generation through a function that accepts the fields string.
- define the writer once.
- define the expected once and use where applicable.

Reply:
Valid point. I am good either way.